src/loader/bgw_launcher.c

```diff
@@ -492,32 +524,43 @@ register_scheduler_handle(VirtualTransactionId vxid, DbHashEntry *db_he)
 static AckResult
 message_start_action(HTAB *db_htab, BgwMessage *message, VirtualTransactionId vxid)
 {
-	DbHashEntry *db_he;
-	pid_t		worker_pid;
+	DbHashEntry *entry;
 
-	db_he = allocate_scheduler(db_htab, message->db_oid);
-	if (db_he == NULL)
-		return ACK_FAILURE;
+	entry = db_hash_entry_create_if_not_exists(db_htab, message->db_oid);
+	entry->vxid = vxid;
 
-	if (get_background_worker_pid(db_he->db_scheduler_handle, &worker_pid) == BGWH_STOPPED)
-		return register_scheduler_handle(vxid, db_he);
+	if (entry->state == DISABLED)
+		scheduler_state_trans_disabled_to_enabled(entry);
 
-	return ACK_SUCCESS;
+	scheduler_state_advance(entry);
+
+	return (entry->state == STARTED ? ACK_SUCCESS : ACK_FAILURE);
 }
 
 static AckResult
 message_stop_action(HTAB *db_htab, BgwMessage *message)
 {
-	DbHashEntry *db_he;
-	bool		found;
+	DbHashEntry *entry;
+
+	entry = db_hash_entry_create_if_not_exists(db_htab, message->db_oid);
```

Comment:
Generally, it seems weird to potentially create  a DB entry on stop action. Isn't it an invariant that a DB entry must exists if we get a stop action or to stop is invalid (or should be ignored)? At the very least I'd like a comment explaining how this is correct?

Reply:
Added a comment for @erimatnor's thing. It needs to be created to put the entry into the DISABLED state because otherwise it will end up in ENABLED on next poll.